### ISSUE TITLE
device.uuid=NULL on IOS 12.3.1

### ISSUE THREAD

**Issue by sitehost** [apache/cordova-plugin-device#103]
## Problem
using  device.uuid on IOS 12.3.1   returns all as NULL

'Device Name: '     + device.name     + '<br />' +
'Device Cordova: '  + device.cordova  + '<br />' +
'Device Platform: ' + device.platform + '<br />' +
'Device UUID: '     + device.uuid     + '<br />' +
'Device Version: '  + device.version  + '<br />';

### What is expected to happen?
on IOS 12.2   device.uuid will display the correct uuid, however on IOS 12.3.1 they return NULL.

### What does actually happen?
device.uuid returns a NULL 

## Information
working from a cordova build, it has worked on all of our devices until we did an update to 12.3.1 . What is the work around, or a solution to gather the IOS uuid 

### Command or Code

run device.uuid on 12.3.1 and on 12.2.1

## Checklist

- [x ] I searched for existing GitHub issues
- [ x] I updated all Cordova tooling to most recent version
- [x ] I included all the necessary information above

**Comment by janpio** [comment]
What would be the "correct uuid"? 
What is the iOS code necessary to get that?

**Comment by sitehost** [comment]
well its not giving any UUID from the device.  it works on ios < 12.1  but on later updates its giving NULL

**Comment by janpio** [comment]
Yes, I understood that. I am asking what is a UUID for the device under iOS 12.3 and what native code Apple provides to get it.

**Comment by sitehost** [comment]
It's a unique device Id. And I am not sure. I'm using this plugin. The code for this plugin to get it is device.uuid in javascript

**Comment by XieXiRong** [comment]
yes, it is a big problem.  not work after ios 12.2

**Comment by XieXiRong** [comment]
> Yes, I understood that.
> I am asking what is a UUID for the device under iOS 12.3 and what native code Apple provides to get it.

just use the method 'device.uuid' and returns null

**Comment by timbru31** [comment]
I can't reproduce this issue.

Is this reproducible in a new, plain Cordova app? A minimal reproduction repository would really help to debug and later fix this issue. More information on how to create one: https://github.com/apache/cordova-contribute/blob/master/create-reproduction.md

**Comment by XieXiRong** [comment]
> yes, it is a big problem. not work after ios 12.2

sorry it is my fault，actually the method is work in ios version 12.2 and 12.3

**Comment by jcesarmobile** [comment]
I can't reproduce neither, even if trying to return `NULL` or `nil` from native makes the app crash instead of returning `NULL`, the only way of returning a null value is by returning `[NSNull null]`, but that returns `null`, not `NULL`.

Anyway, looks like the identifierForVendor can return nil in rare cases, that could cause a crash as mentioned earlier

> If the value is nil, wait and get the value again later. This happens, for example, after the device has been restarted but before the user has unlocked the device.

https://developer.apple.com/documentation/uikit/uidevice/1620059-identifierforvendor

**Comment by cyptus** [comment]
i got the same issue, plugin is not working anymore after ios 12.3.1

**Comment by JoshuvaGeorge03** [comment]
And also device.serial return 'unknown' in ios 12.3.1.. Is any fix for these issue?
